### DISCUSSION TITLE
Weird behaviours when using atomWithDefault with query

### DISCUSSION THREAD

**Discussion by TeaBough** [pmndrs/jotai#1935] [Q&A]
Hi again !

Following the discussion (https://github.com/pmndrs/jotai/discussions/1929) with @dai-shi , I've wrapped my query atom with an `atomWithDefault`.
My problem is now that the the component get "suspended" every time I update the atom wrapped.
Here is the sample code to demonstrate (and the sandbox : https://codesandbox.io/s/with-jotai-2-forked-leejy2?file=/src/App.js)

```ts
const [anAtom] = atomsWithQuery((get) => ({
  queryKey: ["KEY"],
  queryFn: async () => {
    return Promise.resolve({
      field1: 1,
      field2: 1
    });
  }
}));

export const testAtom = atomWithDefault(async (get) => {
  const res = await get(anAtom);
  return res;
});

const field1Atom = selectAtom(testAtom, (p) => p.field1);
const field2Atom = selectAtom(testAtom, (p) => p.field2);

function Field1Display() {
  const field1 = useAtomValue(field1Atom);
  return <div>VALUE FIELD 1 : {field1}</div>;
}

function Field2Display() {
  const field2 = useAtomValue(field2Atom);
  return <div>VALUE FIELD 2 : {field2}</div>;
}

function IncreaseButtons() {
  const setTestAtom = useSetAtom(testAtom);
  return (
    <>
      <button
        onClick={() =>
          setTestAtom((prev) => ({ ...prev, field1: prev.field1 + 1 }))
        }
      >
        Increase field 1
      </button>
      <button
        onClick={() =>
          setTestAtom((prev) => ({ ...prev, field2: prev.field2 + 1 }))
        }
      >
        Increase field 2
      </button>
    </>
  );
}
const App = () => (
  <React.Suspense fallback={<div>LOADING</div>}>
    <Field1Display />
    <Field2Display />
    <IncreaseButtons />
  </React.Suspense>
);
```

Every time I increase `field1` or `field2` I get a `LOADING` displayed and my whole component get's unmounted...
When I increase `field1` I only expect the component `Field1Display` to rerender... how can I achieve that ?

**Comment by dai-shi** [comment]
Nice catch. With current atomWithDefault impl, it's a pitfall. Its signature is a little weird when migrated from v1. I think we can *fix* it, but it can break some usages.

Can you create a reproduction without jotai-query and file a new issue?

The `anAtom` can simply be:
```js
const anAtom = atom((get) => Promise.resolve(...))
```

**Reply by TeaBough** [reply]
Here is the issue : https://github.com/pmndrs/jotai/issues/1937

**Reply by dai-shi** [reply]
Thanks!
